router/forwarder/faulter.go

```diff
@@ -88,19 +88,32 @@ func (self *Faulter) run() {
 				}
 
 				circuitIds := strings.Join(workload, " ")
-				fault := &ctrl_pb.Fault{Subject: ctrl_pb.FaultSubject_ForwardFault, Id: circuitIds}
 
-				log := pfxlog.Logger().WithField("ctrlId", ctrlId)
-				ch := self.ctrls.GetCtrlChannel(ctrlId)
-				if ch == nil {
-					log.Error("no control channel for controller")
-					continue
-				}
-
-				if err := protobufs.MarshalTyped(fault).Send(ch); err == nil {
-					log.WithField("circuitCount", len(workload)).Warn("reported forwarding faults")
-				} else {
-					log.WithError(err).Error("error sending fault report")
+				if ctrlId != "" {
+					log := pfxlog.Logger().WithField("ctrlId", ctrlId)
+					ch := self.ctrls.GetCtrlChannel(ctrlId)
+					if ch == nil {
+						log.Error("no control channel for controller")
+						continue
+					}
+
+					fault := &ctrl_pb.Fault{Subject: ctrl_pb.FaultSubject_ForwardFault, Id: circuitIds}
+					if err := protobufs.MarshalTyped(fault).Send(ch); err == nil {
+						log.WithField("circuitCount", len(workload)).Warn("reported forwarding faults")
+					} else {
+						log.WithError(err).Error("error sending fault report")
```

Comment:
What happens in this condition (we have a specific controller to forward a fault to and can't send it to them)?

I see the fault is dropped. I assume if the destination controller comes back, I assume the circuit w/ the invalid forward is eventually abandoned by the client, the client has made a new circuit (most likely through a new controller), and the fault-containing circuit is eventually cleaned up.

Reply:
If the circuit continues to fault, we'll have another chance to send. If it doesn't, then presumably the issue was resolved in some other way.